whisper_medusa/models/model.py

```diff
@@ -1221,22 +1308,147 @@ def forward(
+                        )
+
+                        if whisper_model_outputs.encoder_last_hidden_state is not None:
+                            whisper_model_outputs.cross_attentions += (
+                                medusa_block_decoder_outputs[2],
+                            )
+
+                    for i in range(len(self.medusa_heads)):
+                        head_out = self.medusa_heads[i](medusa_block_decoder_outputs[0])
+                        head_proj = self.whisper_model.proj_out(head_out)
+                        medusa_logits.append(head_proj)
+                else:
+                    raise ValueError("Invalid medusa_heads_type, received {}".format(self.config.medusa_heads_type))
+            else:
+                if self.config.medusa_heads_type == "medusa_block":
+                    use_cache = (
+                        use_cache if use_cache is not None else self.config.use_cache
+                    )
+                    past_key_value = (
+                        past_key_values[-1] if past_key_values is not None else None
+                    )
+                    medusa_block_decoder_outputs = self.medusa_block(
+                        whisper_model_outputs.last_hidden_state,
+                        attention_mask=attention_mask,
+                        encoder_hidden_states=whisper_model_outputs.encoder_last_hidden_state,
+                        layer_head_mask=(
+                            head_mask[-1] if head_mask is not None else None
+                        ),
+                        cross_attn_layer_head_mask=(
+                            cross_attn_head_mask[-1]
+                            if cross_attn_head_mask is not None
+                            else None
+                        ),
+                        past_key_value=past_key_value,
+                        output_attentions=output_attentions,
+                        use_cache=use_cache,
+                    )
+                    if use_cache:
+                        whisper_model_outputs.past_key_values += (
+                            medusa_block_decoder_outputs[3 if output_attentions else 1],
+                        )
+
+                    if output_attentions:
+                        whisper_model_outputs.decoder_attentions += (
+                            medusa_block_decoder_outputs[1],
+                        )
+
+                        if whisper_model_outputs.encoder_last_hidden_state is not None:
+                            whisper_model_outputs.cross_attentions += (
+                                medusa_block_decoder_outputs[2],
+                            )
+
 
         stack_heads_logits = torch.stack(medusa_logits, dim=0)
 
         loss = None
+        if labels is not None:
+            loss_fct = MedusaCrossEntropyLoss(
+                loss_on_original=self.config.medusa_loss_on_original
+            )
+            if hasattr(self.config, "medusa_kl_loss") and self.config.medusa_kl_loss:
```

Comment:
set the default value to None or false and use:
```suggestion
            if self.config.medusa_kl_loss:
```

Reply:
or you can use:
```suggestion
            if getattr(self.config, "medusa_kl_loss", None):
```

Reply:
done